### PR TITLE
fix(kubernetes): fix manifest update race condition

### PR DESCRIPTION
**Pull request by sbwsg**
Before: Sometimes a user would see parallel Kubernetes deployments merged into a single set of Deploy Statuses in a deployManifest execution. See the bug described in https://github.com/spinnaker/spinnaker/issues/3661

After: the bug is no longer reproducible.  Clicking on a deployment stage will only show the resources deployed as part of that stage.

### CODE REVIEW

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx [hunk 1]
```diff
@@ -122,12 +123,10 @@ export class DeployStatus extends React.Component<IExecutionDetailsSectionProps,
   private saveManifestSubscription(subscription: IManifestSubscription) {
     const idx = this.findSubscriptionIndex(subscription);
     const subscriptions = [...this.state.subscriptions];
-    if (idx === -1) {
-      subscriptions.push(subscription);
```
Comment (sbwsg) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx]: This line was the crux of the problem.  Any received manifest updates were added to the rendered list even if they didn't have anything to do with the currently selected stage.

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx [hunk 2]
```diff
@@ -25,36 +25,36 @@ interface IStageManifest {
...
-
-  public componentDidUpdate() {
-    this.buildSubscriptions();
+  public componentWillReceiveProps(props: IExecutionDetailsSectionProps) {
+    const manifests = this.getStageManifests(props);
+    const manifestIds = manifests.map(m => this.manifestIdentifier(m)).sort();
+    if (manifestIds.join('') !== this.state.manifestIds.join('')) {
+      this.state.subscriptions.forEach(({ unsubscribe }) => unsubscribe());
+      this.setState({ subscriptions: [], manifestIds }, () => {
+        this.buildSubscriptions(props);
```
[the remark this answers is not in the document]
Comment (sbwsg) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx]: Yeah definitely agreed. I'll spend a bit more time on this tomorrow and see if I can untangle it while keeping the behaviour.

[the remark this answers is not in the document]
Comment (christopherthielen) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx]: FYI I think the new `componentWillReceiveProps` hotness is `componentDidUpdate`

[the remark this answers is not in the document]
Comment (sbwsg) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatus.tsx]: I've consolidated into `componentDidUpdate` as suggested.  Much easier to grok now 👍. `setState` is now limited to `componentDidUpdate` and manifest subscription updates.

**Pull request merged**